src/server/game/AI/SmartScripts/SmartScriptMgr.h

```diff
@@ -1479,15 +1473,13 @@ class TC_GAME_API SmartWaypointMgr
 
         void LoadFromDB();
 
-        WPPath* GetPath(uint32 id)
+        WPPath const &GetPath(uint32 id)
         {
-            if (waypoint_map.find(id) != waypoint_map.end())
-                return waypoint_map[id];
-            else return nullptr;
+            return waypoint_map[id];
```

Comment:
http://paste2.org/WGGLIVDX what about this then? reference can be here, even when unordered_map rehashes (adding, deleting) allocated elements will stay in same place. if this approach is no good, then ill just revert to pointer ;p